app/src/main/java/com/concordia/mcga/utilities/pathfinding/PathFinder.java

```diff
@@ -53,6 +56,41 @@ public PathFinder(TiledMap map) {
         return returnList;
     }
 
+    public List<PathFinderTile> shortestPathJunctions(int startX, int startY, int destX, int destY) throws Exception {
+        ArrayList<PathFinderTile> pathTiles = new ArrayList<PathFinderTile>(shortestPath(startX, startY, destX, destY));
+        ArrayList<PathFinderTile> pathTilesJunctions = new ArrayList<PathFinderTile>();
+
+        int curX = 0;
+        int curY = 0;
+        for (int i = 0; i < pathTiles.size(); i++) {
+            if (i == 0) {
+                PathFinderTile pft = pathTiles.get(i);
```

Comment:
This is so ugly... LOL 

Special case `i == 0 ` should be done BEFORE entering the loop... if not the `if(i==0)` gets evaluated at EACH 🐌iteration. 

 Again not Sam... ❤️